modules/geo-layers/src/tile-layer/tile-layer.ts

```diff
@@ -150,15 +161,7 @@ export default class TileLayer<
     return changeFlags.somethingChanged;
   }
 
-  updateState({
-    props,
-    changeFlags
-  }: {
-    props: TileLayerProps;
-    oldProps: TileLayerProps;
-    context: any;
-    changeFlags: ChangeFlags;
-  }) {
+  updateState({changeFlags}: UpdateParameters<ExtraPropsT & Required<_TileLayerProps<DataT>>>) {
```

Comment:
Technically this signature is incorrect as we're missing fact that we also have _CompositeLayerProps included.

It's not used, but that makes me struggle to write proper signature for MVTLayer.

Reply:
MVTLayer signature will have too look like this:
```
updateState({
    props,
    oldProps,
    context,
    changeFlags
  }: UpdateParameters<ExtraPropsT & Required<_MVTLayerProps> & Required<_TileLayerProps<DataT>>>) {
  ```
  
  (that requires to export remember all _parent_, _partial_ parent classes ...)